File: SDKs/amazonaws.com/comprehend/2017-11-27/python/src/sdk/models/shared/entityrecognizermetadata.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class EntityRecognizerMetadata:
    r"""EntityRecognizerMetadata
    Detailed information about an entity recognizer.
    """
    
    entity_types: Optional[List[EntityRecognizerMetadataEntityTypesListItem]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EntityTypes') }})
    evaluation_metrics: Optional[EntityRecognizerEvaluationMetrics] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('EvaluationMetrics') }})
    number_of_test_documents: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('NumberOfTestDocuments') }})
    number_of_trained_documents: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('NumberOfTrainedDocuments') }})
    

File: SDKs/amazonaws.com/comprehend/2017-11-27/typescript/src/sdk/models/shared/entityrecognizermetadata.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { EntityRecognizerMetadataEntityTypesListItem } from "./entityrecognizermetadataentitytypeslistitem";
import { EntityRecognizerEvaluationMetrics } from "./entityrecognizerevaluationmetrics";



// EntityRecognizerMetadata
/** 
 * Detailed information about an entity recognizer.
**/
export class EntityRecognizerMetadata extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=EntityTypes", elemType: EntityRecognizerMetadataEntityTypesListItem })
  entityTypes?: EntityRecognizerMetadataEntityTypesListItem[];

  @SpeakeasyMetadata({ data: "json, name=EvaluationMetrics" })
  evaluationMetrics?: EntityRecognizerEvaluationMetrics;

  @SpeakeasyMetadata({ data: "json, name=NumberOfTestDocuments" })
  numberOfTestDocuments?: number;

  @SpeakeasyMetadata({ data: "json, name=NumberOfTrainedDocuments" })
  numberOfTrainedDocuments?: number;
}
